src/interpreter/map-validator.ts

```diff
@@ -502,11 +480,12 @@ export class MapValidator implements MapAstVisitor {
       const variableName = getVariableName(jessieNode);
       const variable = this.variables[variableName];
 
-      if (variable !== undefined) {
+      if (variable !== undefined && !isInlineCallNode(variable)) {
```

Comment:
What does this change? I don't understand the implications.

Reply:
This will prevent propagation of input errors in `inlineCallNode`. I thought about this and it doesn't solve propagation of errors in other three literal nodes, so I'll refactor it and let you know.

Reply:
Looks like I won't be able to return that helpful errors without propagation of them, so I'll rather wait for type analysis.

For example in this map:
```
map Test {
  f1 = 1
  f2 = 2

  res = {
    f1: f1,
    f2: f2,
  }

  map result res
}
```

with profile:
```
usecase Test {
  result {
    f1! string!
    f2! boolean!
  }
}
```

I can return all these propagated errors:
```
5:18 PrimitiveLiteral - Wrong Structure: expected string, but got 1
8:19 JessieExpression - Wrong Variable Structure: variable f1 expected string!, but got 1
6:18 PrimitiveLiteral - Wrong Structure: expected boolean, but got 2
8:19 JessieExpression - Wrong Variable Structure: variable f2 expected boolean!, but got 2
13:24 JessieExpression - Wrong Variable Structure: variable res expected {f1: string!, f2: boolean!}!, but got {
    f1: f1,
    f2: f2,
  }
```

or keep only last one and let user understand that `f1` and `f2` are not valid types.

Reply:
I would expect that error/warning will show only at the location of map result, but it might be more helpful to have more errors until we don't have type analysis. I'm not sure. @TheEdward162 @Jakub-Vacek what do you think?

Reply:
More errors => higher chance that user will get what is wrong 🤷